scene/3d/physics/soft_body_3d.cpp

```diff
@@ -71,15 +75,70 @@ void SoftBodyRenderingServerHandler::clear() {
 }
 
 void SoftBodyRenderingServerHandler::open() {
-	write_buffer = buffer.ptrw();
+	write_buffer = buffer_curr.ptrw();
 }
 
 void SoftBodyRenderingServerHandler::close() {
 	write_buffer = nullptr;
 }
 
-void SoftBodyRenderingServerHandler::commit_changes() {
-	RS::get_singleton()->mesh_surface_update_vertex_region(mesh, surface, 0, buffer);
+void SoftBodyRenderingServerHandler::fti_pump() {
+	if (buffer_prev.is_empty()) {
```

Comment:
Might be safer to have this condition be if the size is not equal, just in case the editor has changed the size of the soft body, or something like that?

The `resize` function likely does this redundant check for us, but not a lot of harm in doing this explicitly as it should get optimized out.